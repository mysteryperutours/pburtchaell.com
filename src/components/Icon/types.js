export const GITHUB = 'GITHUB';
export const TWITTER = 'TWITTER';
export const FACEBOOK = 'FACEBOOK';
export const DRIBBBLE = 'DRIBBBLE';
