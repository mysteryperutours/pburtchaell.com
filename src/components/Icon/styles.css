:local(.icon) {
  height: .85em;
  width: .85em;
  top: .175rem;
  margin-right: .35em;
  display: inline-block;
  position: relative;
}

a:hover :local(.icon) {
  opacity: .5;
}
