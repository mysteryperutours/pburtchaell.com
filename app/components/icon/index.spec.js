import React from 'react';
import Icon, { types } from './index';

describe(Icon.name, () => {
  it('should render GitHub icon', () => {
    expect(shallow(
      <Icon
        type={types.GITHUB}
      />
    )).toMatchSnapshot();
  });

  it('should render Facebook icon', () => {
    expect(shallow(
      <Icon
        type={types.FACEBOOK}
      />
    )).toMatchSnapshot();
  });

  it('should render Twitter icon', () => {
    expect(shallow(
      <Icon
        type={types.TWITTER}
      />
    )).toMatchSnapshot();
  });

  it('should render Dribbble icon', () => {
    expect(shallow(
      <Icon
        type={types.DRIBBBLE}
      />
    )).toMatchSnapshot();
  });
});
