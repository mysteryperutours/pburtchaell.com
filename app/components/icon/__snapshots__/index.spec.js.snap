exports[`Icon should render Dribbble icon 1`] = `
<span>
  <svg
    enableBackground="new 0 0 20 20"
    version="1.1"
    viewBox="0 0 20 20"
    x="0px"
    xmlns="http://www.w3.org/2000/svg"
    y="0px">
    <path
      d="M9.565,7.421C8.207,5.007,6.754,3.038,6.648,2.893C4.457,3.929,2.822,5.948,2.311,8.38C2.517,8.384,5.793,8.423,9.565,7.421zM10.543,10.061c0.102-0.033,0.206-0.064,0.309-0.094c-0.197-0.447-0.412-0.895-0.637-1.336C6.169,9.843,2.287,9.755,2.15,9.751c-0.003,0.084-0.007,0.166-0.007,0.25c0,2.019,0.763,3.861,2.016,5.252l-0.005-0.006C4.154,15.247,6.304,11.433,10.543,10.061zM5.171,16.194V16.19c-0.058-0.045-0.12-0.086-0.178-0.135C5.099,16.14,5.171,16.194,5.171,16.194zM8.118,2.372C8.111,2.374,8.103,2.376,8.103,2.376c0.006-0.002,0.014-0.002,0.014-0.002L8.118,2.372zM15.189,4.104C13.805,2.886,11.99,2.143,10,2.143c-0.639,0-1.258,0.078-1.852,0.221c0.12,0.16,1.595,2.119,2.938,4.584C14.048,5.839,15.167,4.136,15.189,4.104zM10,19.2c-5.08,0-9.199-4.119-9.199-9.199C0.8,4.919,4.919,0.8,10,0.8c5.082,0,9.2,4.119,9.2,9.201C19.2,15.081,15.082,19.2,10,19.2zM11.336,11.286c-4.611,1.607-6.134,4.838-6.165,4.904c1.334,1.041,3.006,1.666,4.828,1.666c1.088,0,2.125-0.221,3.067-0.621c-0.116-0.689-0.573-3.096-1.679-5.967C11.371,11.274,11.354,11.28,11.336,11.286zM11.69,8.12c0.184,0.373,0.358,0.754,0.523,1.139c0.059,0.135,0.114,0.272,0.17,0.406c2.713-0.342,5.385,0.238,5.473,0.256c-0.019-1.863-0.686-3.572-1.787-4.912C16.051,5.032,14.79,6.852,11.69,8.12zM12.861,10.905c1.031,2.836,1.449,5.142,1.529,5.611c1.764-1.191,3.018-3.08,3.367-5.27C17.601,11.196,15.401,10.499,12.861,10.905z" />
  </svg>
</span>
`;

exports[`Icon should render Facebook icon 1`] = `
<span>
  <svg
    enableBackground="new 0 0 20 20"
    version="1.1"
    viewBox="0 0 20 20"
    x="0px"
    xmlns="http://www.w3.org/2000/svg"
    y="0px">
    <path
      d="M17,1H3C1.9,1,1,1.9,1,3v14c0,1.101,0.9,2,2,2h7v-7H8V9.525h2V7.475c0-2.164,1.212-3.684,3.766-3.684l1.803,0.002v2.605h-1.197C13.378,6.398,13,7.144,13,7.836v1.69h2.568L15,12h-2v7h4c1.1,0,2-0.899,2-2V3C19,1.9,18.1,1,17,1z" />
  </svg>
</span>
`;

exports[`Icon should render GitHub icon 1`] = `
<span>
  <svg
    enableBackground="new 0 0 20 20"
    version="1.1"
    viewBox="0 0 20 20"
    x="0px"
    xmlns="http://www.w3.org/2000/svg"
    y="0px">
    <path
      d="M13.18,11.309c-0.718,0-1.3,0.807-1.3,1.799c0,0.994,0.582,1.801,1.3,1.801s1.3-0.807,1.3-1.801C14.479,12.116,13.898,11.309,13.18,11.309zM17.706,6.626c0.149-0.365,0.155-2.439-0.635-4.426c0,0-1.811,0.199-4.551,2.08c-0.575-0.16-1.548-0.238-2.519-0.238c-0.973,0-1.945,0.078-2.52,0.238C4.74,2.399,2.929,2.2,2.929,2.2C2.14,4.187,2.148,6.261,2.295,6.626C1.367,7.634,0.8,8.845,0.8,10.497c0,7.186,5.963,7.301,7.467,7.301c0.342,0,1.018,0.002,1.734,0.002c0.715,0,1.392-0.002,1.732-0.002c1.506,0,7.467-0.115,7.467-7.301C19.2,8.845,18.634,7.634,17.706,6.626zM10.028,16.915H9.972c-3.771,0-6.709-0.449-6.709-4.115c0-0.879,0.31-1.693,1.047-2.369c1.227-1.127,3.305-0.531,5.662-0.531c0.01,0,0.02,0,0.029,0c0.01,0,0.018,0,0.027,0c2.357,0,4.436-0.596,5.664,0.531c0.735,0.676,1.045,1.49,1.045,2.369C16.737,16.466,13.8,16.915,10.028,16.915zM6.821,11.309c-0.718,0-1.3,0.807-1.3,1.799c0,0.994,0.582,1.801,1.3,1.801c0.719,0,1.301-0.807,1.301-1.801C8.122,12.116,7.54,11.309,6.821,11.309z" />
  </svg>
</span>
`;

exports[`Icon should render Messenger icon 1`] = `
<span>
  <svg
    enableBackground="new 0 0 20 20"
    version="1.1"
    viewBox="0 0 20 20"
    x="0px"
    xmlns="http://www.w3.org/2000/svg"
    y="0px">
    <path
      d="M9.9,0C4.4,0,0,4.1,0,9.3c0,2.9,1.4,5.5,3.7,7.2v0V20L7,18.1c0.9,0.3,1.9,0.4,2.9,0.4c5.5,0,9.9-4.1,9.9-9.3C19.8,4.1,15.4,0,9.9,0zM10.9,12.4L8.4,9.7l-4.9,2.7l5.4-5.7l2.6,2.6l4.8-2.6L10.9,12.4z" />
  </svg>
</span>
`;

exports[`Icon should render Twitter icon 1`] = `
<span>
  <svg
    enableBackground="new 0 0 20 20"
    version="1.1"
    viewBox="0 0 20 20"
    x="0px"
    xmlns="http://www.w3.org/2000/svg"
    y="0px">
    <path
      d="M17.316,6.246c0.008,0.162,0.011,0.326,0.011,0.488c0,4.99-3.797,10.742-10.74,10.742c-2.133,0-4.116-0.625-5.787-1.697c0.296,0.035,0.596,0.053,0.9,0.053c1.77,0,3.397-0.604,4.688-1.615c-1.651-0.031-3.046-1.121-3.526-2.621c0.23,0.043,0.467,0.066,0.71,0.066c0.345,0,0.679-0.045,0.995-0.131c-1.727-0.348-3.028-1.873-3.028-3.703c0-0.016,0-0.031,0-0.047c0.509,0.283,1.092,0.453,1.71,0.473c-1.013-0.678-1.68-1.832-1.68-3.143c0-0.691,0.186-1.34,0.512-1.898C3.942,5.498,6.725,7,9.862,7.158C9.798,6.881,9.765,6.594,9.765,6.297c0-2.084,1.689-3.773,3.774-3.773c1.086,0,2.067,0.457,2.756,1.191c0.859-0.17,1.667-0.484,2.397-0.916c-0.282,0.881-0.881,1.621-1.66,2.088c0.764-0.092,1.49-0.293,2.168-0.594C18.694,5.051,18.054,5.715,17.316,6.246z" />
  </svg>
</span>
`;
